File: python_sdk/src/reality_capture/specifications/eval_o2d.py
from pydantic import BaseModel, Field
from typing import Optional
from enum import Enum


class EvalO2DInputs(BaseModel):
    reference: str = Field(description="Reality data id of ContextScene, annotated with embedded 2D object references")
    prediction: str = Field(description="Reality data id of ContextScene, "
                                        "annotated with embedded 2D object predictions")


class EvalO2DOutputs(BaseModel):
    report: str = Field(description="Reality data id of json report with binary classification.")
    objects2d: str = Field(alias="objects2D", description="Reality data id of ContextScene, "
                                                          "annotated with classified embedded 2D objects")


class EvalO2DOutputsCreate(Enum):
    REPORT = "report"
    OBJECTS2D = "objects2D"


class EvalO2DOptions(BaseModel):
    threshold_iou: Optional[int] = Field(None, alias="thresholdIOU", description="Intersection over union threshold")


class EvalO2DSpecificationsCreate(BaseModel):
    inputs: EvalO2DInputs = Field(description="Inputs")
    outputs: list[EvalO2DOutputsCreate] = Field(description="Outputs")
    options: Optional[EvalO2DOptions] = Field(None, description="Options")


class EvalO2DSpecifications(BaseModel):
    inputs: EvalO2DInputs = Field(description="Inputs")
    outputs: EvalO2DOutputs = Field(description="Outputs")
    options: Optional[EvalO2DOptions] = Field(None, description="Options")

File: typescript_sdk/reality_capture/src/specifications/eval_o2d.ts
import { z } from "zod";

export const EvalO2DInputsSchema = z.object({
  reference: z.string().describe("Reality data id of ContextScene, annotated with embedded 2D object references"),
  prediction: z.string().describe("Reality data id of ContextScene, annotated with embedded 2D object predictions"),
});
export type EvalO2DInputs = z.infer<typeof EvalO2DInputsSchema>;

export const EvalO2DOutputsSchema = z.object({
  report: z.string()
    .regex(/^bkt:.+/)
    .describe("Path in Bucket of json report with binary classification.")
    .optional(),
  objects2D: z.string()
    .describe("Reality data id of ContextScene, annotated with classified embedded 2D objects")
    .optional(),
});
export type EvalO2DOutputs = z.infer<typeof EvalO2DOutputsSchema>;

export enum EvalO2DOutputsCreate {
  REPORT = "report",
  OBJECTS2D = "objects2D",
}

export const EvalO2DOptionsSchema = z.object({
  thresholdIOU: z.number()
    .describe("Intersection over union threshold")
    .optional(),
});
export type EvalO2DOptions = z.infer<typeof EvalO2DOptionsSchema>;

export const EvalO2DSpecificationsCreateSchema = z.object({
  inputs: EvalO2DInputsSchema.describe("Inputs"),
  outputs: z.array(z.nativeEnum(EvalO2DOutputsCreate)).describe("Outputs"),
  options: EvalO2DOptionsSchema.describe("Options").optional(),
});
export type EvalO2DSpecificationsCreate = z.infer<typeof EvalO2DSpecificationsCreateSchema>;

export const EvalO2DSpecificationsSchema = z.object({
  inputs: EvalO2DInputsSchema.describe("Inputs"),
  outputs: EvalO2DOutputsSchema.describe("Outputs"),
  options: EvalO2DOptionsSchema.describe("Options").optional(),
});
export type EvalO2DSpecifications = z.infer<typeof EvalO2DSpecificationsSchema>;